app/data_models/questionnaire_store.py

```diff
@@ -58,7 +59,7 @@ def serialize(self):
             "PROGRESS": self.progress_store.serialize(),
             "RESPONSE_METADATA": self.response_metadata,
         }
-        return json.dumps(data, for_json=True)
+        return json.dumps(data, default=for_json, number_mode=json.NM_DECIMAL)
```

Comment:
I think we should consider introducing a wrapper method for loading and dumping JSON that handles the `default` and `number_mode`. Something like:

```python
def load_json(data):
    return json.loads(data, number_mode=json.NM_DECIMAL)

def dump_json(data):
    return json.dumps(data, default=for_json, number_mode=json.NM_DECIMAL)
```

This would also abstract away the JSON library, so it could be easily swapped out.